src/language/visitor.ts

```diff
@@ -1,41 +1,179 @@
-const estraverse = require('estraverse');
-const esprima = require('esprima');
+import * as parser from '@babel/parser';
+import traverse from '@babel/traverse';
 
-export type CompletionNodes = {
+export type CompletionState = {
   databaseName: string | null;
   collectionName: string | null;
   isObjectKey: boolean;
-  isMemberExpression: boolean;
+  isShellMethod: boolean;
   isUseCallExpression: boolean;
   isDbCallExpression: boolean;
-  dbCallPosition: { line: number; character: number };
+  isCollectionName: boolean;
   isAggregationCursor: boolean;
   isFindCursor: boolean;
 };
 
 export class Visitor {
-  constructor() {}
+  _connection: any;
+  _state: CompletionState;
+  _position: { line: number; character: number };
+  _placeholder: string;
+
+  constructor() {
+    this._state = this.getDefaultNodesValues();
+    this._position = { line: 0, character: 0 };
+    this._placeholder = 'TRIGGER_CHARACTER';
```

Comment:
Does this placeholder change? If not maybe we can make it a constant.

Reply:
Does not change. Totally makes sense to make it a constant! Updated.